src/modules/emote_menu/youtube/EmoteMenu.jsx

```diff
@@ -90,15 +91,16 @@ export default class EmoteMenuModule {
   }
 
   show(visible) {
-    // TODO: take into account emote menu setting in the future
+    const emoteMenuEnabled = settings.get(SettingIds.EMOTE_MENU);
+
     const nativeContainer = document.querySelector(NATIVE_EMOTE_MENU_BUTTON_CONTAINER_SELECTOR);
     if (nativeContainer != null) {
-      nativeContainer.classList.toggle(styles.hideEmoteMenuButton, visible);
+      nativeContainer.classList.toggle(styles.hideEmoteMenuButton, visible && emoteMenuEnabled);
```

Comment:
can you explain these changes? why accept `visible` at all if we're simply overwriting its value? why are we getting the setting from storage twice in this path (load -> settings.get -> show -> settings.get)?

Reply:
This is copied logix from twitch/Settings.jsx, I'm unsure why we even call this.show(true) at the end of renderSettings() as we call this.show(true) anyway in the onMount prop on line 39.

was this final this.show(true) a fail-safe?